src/Components/Trainers/FormTrainers/index.jsx

```diff
@@ -1,90 +1,56 @@
-import React from 'react';
-import Button from '../../Shared/Button';
-import { useState, useEffect } from 'react';
 import { Link, useParams, useHistory } from 'react-router-dom';
+import { useState, useEffect } from 'react';
 import { Input } from '../../Shared/Inputs';
+import { sendTrainer, putTrainer } from '../../../redux/trainers/thunks';
+import { useSelector, useDispatch } from 'react-redux';
+import React from 'react';
+import Button from '../../Shared/Button';
 import Modal from '../../Shared/Modal';
 import styles from './form.module.css';
 
 const TrainerAddForm = () => {
   const [successAddModal, setSuccessAddModal] = useState(false);
   const [successEditModal, setSuccessEditModal] = useState(false);
-  const [inputs, setInputs] = useState({});
+  const [inputs, setInputs] = useState({
+    firstName: '',
+    lastName: '',
+    dni: '',
+    phone: '',
+    email: '',
+    city: '',
+    password: '',
+    salary: ''
+  });
   const { id } = useParams();
+  const { trainers } = useSelector((state) => state.trainers);
+  const dispatch = useDispatch();
   const history = useHistory();
-  const onRedirect = {
-    pathname: '/trainers',
-    state: { message: '' }
-  };
+  // const onRedirect = {
+  //   pathname: '/trainers',
+  //   state: { message: '' }
+  // };
   useEffect(() => {
     if (id) {
-      getTrainerID(id);
+      const trainer = trainers.find((trainer) => trainer._id === id);
+      setInputs(trainer);
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
-  const getTrainerID = async (id) => {
-    try {
-      const res = await fetch(`${process.env.REACT_APP_API_URL}/api/trainers/${id}`);
-      const { data } = await res.json();
-      if (data) {
-        setInputs(data);
-      }
-      return data;
-    } catch (error) {
-      console.error(error);
-    }
-  };
-  const putTrainer = async (id, item) => {
-    const setEditTrainer = {
-      firstName: item.firstName,
-      lastName: item.lastName,
-      dni: item.dni.toString(),
-      phone: item.phone.toString(),
-      email: item.email,
-      city: item.city,
-      password: item.password,
-      salary: item.salary.toString()
-    };
-    try {
-      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/trainers/${id}`, {
-        method: 'PUT',
-        body: JSON.stringify(setEditTrainer),
-        headers: {
-          'Content-Type': 'application/json'
-        }
-      });
-      const { message, error, data } = await response.json();
-      if (!error) {
-        console.log(message);
-      } else {
-        throw message;
-      }
+  useEffect(() => {
+    if (successAddModal) {
```

Comment:
This logic its not usefull, you shouldn't handle the success with the modal, you should make a property on the global state to handle this.

Reply:
It is not necessary to handle it with a global state, I think it is valid to do it this way.

Reply:
But it's not working